### PR TITLE
fix(helm): update helm and traefik

### PR DESCRIPTION
**Pull request by DanielHabenicht**
make the demo work again :)

### CODE REVIEW

File: Phonebook/readme.md [hunk 1]
```diff
@@ -63,9 +63,9 @@ helm delete <deployment-name>
 
 ## Development
 
-You need `helm2` and a running kubernetes cluster.
+You need `helm3` and a running kubernetes cluster.
 
-1. Add `phonebook-demo.local` to your hosts, pointing to `localhost` (127.0.0.1).
+1. Add `phonebook-demo.local` to your hosts, pointing to `localhosts` (127.0.01).
```
Comment (DanielHabenicht) [inline Phonebook/readme.md:68]:
```suggestion
1. Add `phonebook-demo.local` to your hosts, pointing to `localhost` (127.0.0.1).
```

File: demo/assets/Dockerfile [hunk 2]
```diff
@@ -6,6 +6,6 @@ COPY ./nginx/ ./
 
 
 RUN rm -r /usr/share/nginx/html/*
-COPY ./assets /usr/share/nginx/html
+COPY ./assets /usr/share/nginx/html/assets
```
Comment (DanielHabenicht) [inline demo/assets/Dockerfile:9]:
```suggestion
COPY ./assets /usr/share/nginx/html
```

File: .azure/pipelines/Phonebook.preview.yml [hunk 3]
```diff
@@ -93,9 +93,7 @@ jobs:
         artifact: $(helm_artifact_name)
 
       - task: HelmInstaller@1
-        displayName: 'Install Helm 2.14.2'
-        inputs:
-          helmVersionToInstall: 2.14.2
```
Comment (paule96) [inline .azure/pipelines/Phonebook.preview.yml:98]: What does the task if you remove the version? install the latest? Can this break something?`should we maybe use something like 2.x.x?

File: Phonebook/local-values.yml [hunk 4]
```diff
@@ -19,14 +19,33 @@ source:
 
 traefik:
   enabled: true
-  startupArguments:
-    - ''
+  additional:
+    checkNewVersion: false
+    sendAnonymousUsage: false
```
Comment (paule96) [inline Phonebook/local-values.yml:24]: is `local-values.yml` for the preview? Can we activate in the preview the anonymous usage report to help the traffic developers?  I don't see why we must disable this in the preview.

File: Phonebook/local-values.yml [hunk 5]
```diff
@@ -19,14 +19,33 @@ source:
...
+    # Enable the dashboard on Traefik
+    enable: true
+    # Expose the dashboard and api through an ingress route at /dashboard
+    # and /api This is not secure and SHOULD NOT be enabled on production
+    # deployments
+    ingressRoute: true
+
+  additionalArguments:
+    - '--providers.kubernetesingress'
+    - '--api.insecure=true'
```
Comment (paule96) [inline Phonebook/local-values.yml:35]: Question: `api.insecure` that the APIs behind the traefik proxy can be http or https with an invalid cert?

File: Phonebook/phonebook/requirements.yaml [hunk 6]
```diff
@@ -1,6 +1,6 @@
 dependencies:
   - name: traefik
     condition: traefik.enabled
-    version: 1.86.2
-    appVersion: 1.7.14
-    repository: https://kubernetes-charts.storage.googleapis.com
+    version: 3.4.0
```
Comment (paule96) [inline Phonebook/phonebook/requirements.yaml:4]: does the yml support wildcards for version numbers? so we can do something like `3.4.x` or `3.x.x`. So we get the security fixes faster?

**Review by paule96 — CHANGES_REQUESTED** [review]

File: .azure/pipelines/Phonebook.preview.yml (hunk 3 above)
Comment (DanielHabenicht) [reply .azure/pipelines/Phonebook.preview.yml:98]: nope, we updated to the newest and we will automatically update until it breaks again :) 
This way we know if it breaks, and then we can fix the version again.

File: Phonebook/local-values.yml (hunk 4 above)
Comment (DanielHabenicht) [reply Phonebook/local-values.yml:24]: Nope, it is for local development only.

File: Phonebook/local-values.yml (hunk 5 above)
Comment (DanielHabenicht) [reply Phonebook/local-values.yml:35]: thats only for local development and is there in order to show the traefik dashboard.

File: Phonebook/phonebook/requirements.yaml (hunk 6 above)
Comment (DanielHabenicht) [reply Phonebook/phonebook/requirements.yaml:4]: nope

**Review by paule96 — APPROVED** [review]

**Pull request merged**